src/EditorFeatures/Core/Classification/Syntactic/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -243,8 +245,8 @@ public void DisconnectFromWorkspace()
 
             if (_workspace != null)
             {
-                _workspace.WorkspaceChanged -= this.OnWorkspaceChanged;
-                _workspace.DocumentActiveContextChanged -= this.OnDocumentActiveContextChanged;
+                _workspaceChangedDisposer?.Dispose();
+                _workspaceDocumentActiveContextChangedDisposer?.Dispose();
```

Comment:
nit: could pull out of the 'if'.  but seems entirely fine to do it this way.  we could also null these out, like nulling out the workspace.